File: petshop.py
from datetime import datetime

def calcular_preco_total(data, qtd_pequenos, qtd_grandes):
 
    petshops = {
        "Meu Canino Feliz": {"distancia": 2, "preco_semana_pequeno": 20, "preco_semana_grande": 40, "aumento_fim_de_semana": 0.2},
        "Vai Rex": {"distancia": 1.7, "preco_semana_pequeno": 15, "preco_semana_grande": 50, "aumento_fim_de_semana_pequeno": 20, "aumento_fim_de_semana_grande": 5},
        "ChowChawgas": {"distancia": 0.8, "preco_pequeno": 30, "preco_grande": 45}
    }

    
    data_formatada = datetime.strptime(data, "%d/%m/%Y")

    fim_de_semana = data_formatada.weekday() >= 5

    melhor_petshop = None
    menor_preco_total = float('inf')

    for petshop, info in petshops.items():
        preco_pequeno = info["preco_pequeno"] if "preco_pequeno" in info else info["preco_semana_pequeno"]
        preco_grande = info["preco_grande"] if "preco_grande" in info else info["preco_semana_grande"]

        if fim_de_semana:
            preco_pequeno += preco_pequeno * info["aumento_fim_de_semana_pequeno"] / 100
            preco_grande += preco_grande * info["aumento_fim_de_semana_grande"] / 100

        preco_total = qtd_pequenos * preco_pequeno + qtd_grandes * preco_grande

        if preco_total < menor_preco_total or (preco_total == menor_preco_total and info["distancia"] < melhor_petshop["distancia"]):
            melhor_petshop = {"nome": petshop, "distancia": info["distancia"], "preco_total": preco_total}
            menor_preco_total = preco_total

    return melhor_petshop

entrada = "03/08/2018 3 5"
data, qtd_pequenos, qtd_grandes = entrada.split()[0], int(entrada.split()[1]), int(entrada.split()[2])

resultado = calcular_preco_total(data, qtd_pequenos, qtd_grandes)

print(f"Melhor petshop: {resultado['nome']}")
print(f"Preço total dos banhos: R${resultado['preco_total']:.2f}")

File: Projeto_Dti/src/Projeto_Dti/petshop.java

package Projeto_Dti;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;

public class petshop {

    public static void main(String[] args) {
    	
        String entrada = "03/08/2018 3 5";
        String[] partesEntrada = entrada.split(" ");
        Date data;
        int qtdPequenos, qtdGrandes;

        try {
            data = new SimpleDateFormat("dd/MM/yyyy").parse(partesEntrada[0]);
            qtdPequenos = Integer.parseInt(partesEntrada[1]);
            qtdGrandes = Integer.parseInt(partesEntrada[2]);
        } catch (ParseException e) {
            System.out.println("Erro ao processar a entrada.");
            return;
        }

        double precoTotalMeuCaninoFeliz = calcularPrecoTotal(2.0, data, qtdPequenos, qtdGrandes, 20.0, 40.0, 0.2);
        double precoTotalVaiRex = calcularPrecoTotal(1.7, data, qtdPequenos, qtdGrandes, 15.0, 50.0, 0.2);
        double precoTotalChowChawgas = calcularPrecoTotal(0.8, data, qtdPequenos, qtdGrandes, 30.0, 45.0, 0.0);

        String melhorPetshop = "";
        double menorPrecoTotal = Double.MAX_VALUE;

        if (precoTotalMeuCaninoFeliz < menorPrecoTotal) {
            melhorPetshop = "Meu Canino Feliz";
            menorPrecoTotal = precoTotalMeuCaninoFeliz;
        }

        if (precoTotalVaiRex < menorPrecoTotal) {
            melhorPetshop = "Vai Rex";
            menorPrecoTotal = precoTotalVaiRex;
        }

        if (precoTotalChowChawgas < menorPrecoTotal) {
            melhorPetshop = "ChowChawgas";
        }

        System.out.println("Melhor petshop: " + melhorPetshop);
        System.out.println("Preço total dos banhos: R$" + menorPrecoTotal);
    }

    private static double calcularPrecoTotal(double distancia, Date data, int qtdPequenos, int qtdGrandes,
            double precoPequeno, double precoGrande, double aumentoFimDeSemana) {
        boolean fimDeSemana = Calendar.DAY_OF_WEEK == 0 || Calendar.DAY_OF_WEEK == 6;
        double precoFinalPequeno = fimDeSemana ? precoPequeno * (1 + aumentoFimDeSemana) : precoPequeno;
        double precoFinalGrande = fimDeSemana ? precoGrande * (1 + aumentoFimDeSemana) : precoGrande;

        return qtdPequenos * precoFinalPequeno + qtdGrandes * precoFinalGrande;
    }
}


